### PR TITLE
Results section, GPU performance plots

### PR DESCRIPTION
**Pull request by armansito**
Still needs a little bit of discussion. Also needs a section on encoding and bump estimation. I'm planning to push more changes tomorrow but feel free to start reviewing what I have so far.

(Setting base to `mmark-120k` to keep the delta clean)

### CODE REVIEW

File: paper.tex [hunk 1]
```diff
@@ -474,11 +466,55 @@ \section{Performance}
...
+
+We evaluated the runtime performance of our compute shader on 4 GPU models and multiple form factors: Google Pixel 6 equipped with the Arm \emph{Mali-G78 MP20} mobile GPU, the Apple \emph{M1 Max} integrated laptop GPU, as well as two discrete desktop GPUs: the mid-2015 NVIDIA {GeForce GTX 980Ti} and the late-2022 NVIDIA {GeForce RTX 4090}. We authored our entire GPU pipeline in the \emph{WebGPU Shading Language} (WGSL) and used the \emph{wgpu}\cite{wgpu} framework to natively interoperate with the native graphics APIs (we tested the M1 Max on macOS via the Metal API; the mobile and desktop GPUs were tested via the Vulkan API on Android and Ubuntu systems).
+
+We instrumented our pipeline to obtain fine-grained pipeline stage execution times using the GPU \emph{timestamp query} feature supported by both Metal and Vulkan. This allowed us to collect measurements for the curve flattening stage in isolation. We ran the compute pipeline on the \emph{timings} SVG files provided by Nehab2020. The largest of these SVGs is \emph{waves} (see rendering \emph{(a)} in Figure \ref{fig:renderings}) which contains 42 stroked paths and a total of 13,308 path segments.
```
Comment (raphlinus) [inline paper.tex]: Should be `\cite{Nehab2020}`

File: paper.tex [hunk 2]
```diff
@@ -474,11 +466,55 @@ \section{Performance}
...
+\begin{figure}
+    \includegraphics[scale=0.65]{nehab_gpu_timings}
+    \caption{Mean GPU execution times of the compute shader that generates stroked and filled outlines, for the Nehab2020 \emph{timings} data set, in microseconds. The hatched bars show the timings for the version that outputs line primitives while the solid bars show that for arcs. Mali-G78 (the lowest end GPU we tested) converts \emph{waves} (the largest scene in the set) to lines in ~3ms. The compute shader executes at least an order of magnitude faster on the other GPUs.}
+    \Description{A stacked bar chart of the GPU timings for stroke expansion from Nehab2020 timings SVGs}
+    \label{fig:nehab-gpu-timings}
+\end{figure}
+
+\begin{figure}
+    \includegraphics[scale=0.66]{test_scenes_gpu_timings}
+    \caption{Mean GPU execution times of the compute shader for the stress test scenes.}
```
Comment (raphlinus) [inline paper.tex]: The caption should explicitly state the task - which I believe is stroke expansion and flattening here. The description does, but my understanding is that it is not generally shown in the printed copy and is for accessibility.

File: paper.tex [hunk 3]
```diff
@@ -392,15 +386,9 @@ \subsection{Pipeline Design} \label{subsec:pipeline-design}
...
-%         * Parallel lines per ES makes sense but parallel arcs per ES much less so, just because of the prim count. Thus, such detailed discussion may make sense downstream of the prim count results.
+A property of this data-parallel structure is that the output is unordered. The cubic-Bézier-to-Euler-spiral conversion is sequential within a single thread, so it is possible to preserve ordering at the scope of an individual path segment, if desired. However, ordering cannot be guaranteed across path segments. We refer to this output data structure as \emph{line soup}. We integrated the compute shader as a stage in a larger compute shader pipeline (discussed in Section \ref{subsection:gpu-results}) that spatially sorts the line soup in subsequent binning and tiling stages prior to rasterization.
 
+The compute shader writes the line soup output to a GPU storage buffer. Each line soup entry contains the viewport coordinates of the two endpoints of a singe line segment. We also implemented a variant that outputs circular arc segments instead of lines, in which the output entries contain the arc endpoints in addition to its curvature.
```
Comment (raphlinus) [inline paper.tex]: I think it's worth mentioning here that it also includes the path id, which in turn can be used to retrieve the paint for the path.

File: paper.tex [hunk 4]
```diff
@@ -444,21 +432,25 @@ \subsection{Flattening Algorithm}
...
-\section{Performance}
+\section{Results}
 
 % TODO: We integrated our algorithm into a fully GPU-driven rendering pipeline, Vello WGSL
 % TODO: 3 alternative implementations: cubic->ES->LineSoup (current), cubic->ES->ArcSoup, less importantly: cubic->ES (for load balancing eval).
 % TODO: Take a measurement with ablated bbox atomic
 % TODO: discuss: GPUs tested, data sets. Test strategy
 % TODO: discuss: divergence (reference prim count)
 
+We implemented two versions of our stroking method: a sequential CPU stroker and the GPU compute shader outlined in Section \ref{section:gpu-impl}. Both versions can generate stroked outlines with line and arc primitives. We focused our evaluation on the number of output primitives and execution time, using the \emph{timings} dataset provided by Nehab2020\cite{Nehab2020}, in addition to our own curve-intensive scenes to stress the GPU implementation. We presents our findings in the remainder of this section.
```
Comment (raphlinus) [inline paper.tex]: "We present"

**Review by raphlinus — APPROVED** [review]
Great to get this in. Minor suggestions inline.

File: paper.tex (hunk 1 above)
Comment (armansito) [reply paper.tex]: Going with `Nehab\cite{Nehab2020}` per discussion.

File: paper.tex (hunk 3 above)
Comment (armansito) [reply paper.tex]: Done. Note that the Input Encoding section also mentions this but it doesn't hurt to mention it here.

File: paper.tex (hunk 4 above)
Comment (armansito) [reply paper.tex]: Done.

**Pull request merged**